src/participant.rs

```diff
@@ -250,6 +254,10 @@ impl Identity {
         Self::new(verification_key, encryption_key, signature)
             .map_err(|e| io::Error::new(io::ErrorKind::Other, e))
     }
+
+    pub fn verify_data(&self, data: &[u8], signature: &Signature) -> Result<(), SignatureError> {
+        self.verification_key.verify(data, &signature)
```

Comment:
Nit: this looks like a double-reference that will be automatically dereferenced. `&signature` -> `signature` should work.

This is the kind of things that `cargo clippy` would catch, which reminds me we need to run it as part of the GitHub CI.

Reply:
Fixed, thanks